mobile/src/employee-details/employee-details-events-list.tsx

```diff
@@ -43,9 +46,34 @@ export class EmployeeDetailsEventsList extends Component<EmployeeDetailsEventsLi
                     <View style={eventRow}>
                         <ApplicationIcon name={this.eventTypeToGlyphIcon.get(item.type)} style={eventIcon} />
                         <StyledText style={eventTitle}>{item.type} starts on {item.dates.startDate.format(eventDialogTextDateFormat)} and completes on {item.dates.endDate.format(eventDialogTextDateFormat)} ({item.status})</StyledText>
+                        {
+                            this.eventManagementControls(item.calendarEventId)
+                        }
                     </View>
                 </View>
         );
     }
 
+    private eventManagementControls(calendarEventId: string) {
+        return (this.props.events.find(e => e.calendarEventId === calendarEventId).status === CalendarEventStatus.Requested) ? 
+        <View style={{ paddingLeft: 10, flexDirection: 'column', alignItems: 'center' }}>
+            <TouchableOpacity 
+                onPress={() => this.updateCalendarEvent(calendarEventId, CalendarEventStatus.Approved)}>
+                <Text style={{ fontSize: 9, color: 'green' }}>APPROVE</Text>
+            </TouchableOpacity>
+            <TouchableOpacity 
+                onPress={() => this.updateCalendarEvent(calendarEventId, CalendarEventStatus.Rejected)}>
+                    <Text style={{ fontSize: 9, color: 'red' }}>REJECT</Text>
+            </TouchableOpacity>
+        </View> : null;
+    }
+
+    private updateCalendarEvent(calendarEventId: string, status: CalendarEventStatus) {
+        const event = this.props.events.find(e => e.calendarEventId === calendarEventId);
```

Comment:
why the `find` used several time there and at line #58 ? Event should be passed to the method